README.md

```diff
@@ -0,0 +1,105 @@
+
+## Deployment with Terraform
+
+### Initializing
+
+From the repository's main folder, Terraform's [`init`](https://www.terraform.io/docs/cli/commands/init.html) command will prepare the configuration:
+
+```bash
+terraform init
+```
+
+### Planning
+
+A [`plan`](https://www.terraform.io/docs/cli/commands/plan.html) can then be created.
+
+The following variables are required:
+
+* `azure_tenant_id`: the Azure tenant under which the subscription and application are located
+* `azure_subscription_id`: the subscription where the Vault cluster is to be deployed
+* `azure_client_id`: the application's service principal with necessary permissions for the deployment
+* `azure_client_secret`: the service principal's secret password
+* `vault_public_domain_name`: the application's ID URI
+* `vault_source_image_id`: the virtual machine image set to run the `vault` service
+
+The following example saves the plan into the `vault.plan` file:
+
+```bash
+terraform plan \
+  -var azure_tenant_id=d2e32f1f-2b06-4170-8082-e44928f950df \
+  -var azure_subscription_id=be45123f-2769-4b5a-a7d1-5d3772b11a8b \
+  -var azure_client_id=747a4502-4be3-4619-9936-717683af7592 \
+  -var azure_client_secret=secret \
+  -var vault_public_domain_name='my-app'
+  -var vault_source_image_id=/subscriptions/be45123f-2769-4b5a-a7d1-5d3772b11a8b/resourceGroups/my-resource-group/providers/Microsoft.Compute/images/vault-image \
+  -out vault.plan
+```
+
+### Deploying
+
+With the deployment planned, Terraform's [`apply`](https://www.terraform.io/docs/cli/commands/apply.html) command will trigger the deployment:
+
+```bash
+terraform apply vault.plan
+```
+
+## Post-Deployment
+
+There are some things that need to be done after the Terraform deployment.
+
+If SSH access is needed to one of the VM, the IP address and port can be found in the Load Balancer `vault-lb`, under the Inbound NAT Rules section. For example:
+
+```bash
+ssh vm-user@12.34.56.78 -p 50001
+```
+
+### Initializing the Vault Cluster
+
+All Vault instances are currently non-initialized and sealed. One of the instances should be initialized, which will automatically unseal it. The other instances will join and form a cluster.
+
+```bash
+export VAULT_IP_ADDR="http://$(terraform output -raw lb_ip_addr):8200"
+
+# Initializes one of the Vault instances (through the load balancer). This will
+# in turn unseal all other Vault instances in the cluster.
+export ROOT_TOKEN=$(curl --request PUT http://${VAULT_IP_ADDR}:8200/v1/sys/init --data '{"secret_shares":5,"secret_threshold":3,"recovery_shares":5,"recovery_threshold":3}' --header 'Content-Type:application/json' --silent | jq -r '.root_token')
+```
+
+TODO: What to do with the secret and recovery tokens? They are needed to regenerate a root token.
+
+### Strengthening for Production
```

Comment:
Is the intended target audience technical enough to perform (and understand!) these steps?

Reply:
I hope so, if you use a TF templates, you should understand the Hashicorp's documentation